llama_stack/providers/remote/inference/vllm/vllm.py

```diff
@@ -351,17 +350,41 @@ async def _get_model(self, model_id: str) -> Model:
             raise ValueError("Model store not set")
         return await self.model_store.get_model(model_id)
 
-    def _lazy_initialize_client(self):
-        if self.client is not None:
-            return
+    def _get_client(self) -> AsyncOpenAI:
+        """Get vLLM client with dynamic authentication token support."""
+        # Try to get authentication from provider data first
+        provider_data = self.get_request_provider_data()
+
+        api_token = None
+        url = None
+
+        if provider_data:
+            api_token = getattr(provider_data, "vllm_api_token", None)
+            url = getattr(provider_data, "vllm_url", None)
 
-        log.info(f"Initializing vLLM client with base_url={self.config.url}")
-        self.client = self._create_client()
+        # Fall back to config if no provider data
+        if not api_token:
+            api_token = self.config.api_token
+        if not url:
+            url = self.config.url
+
+        if not api_token:
```

Comment:
I think a better error message would be: "Please provide the API Token either as part of the provider configuration or pass it in the header as ..." ?